### PR TITLE
Cleanup and build fixes with Hikey Android

### PR DESCRIPTION
**Pull request by kuscsik**

### CODE REVIEW

**Comment by kuscsik** [comment]
CI failed also on previous patches with same error:

https://github.com/OP-TEE/optee_client/pull/47#partial-pull-merging
https://travis-ci.org/OP-TEE/optee_client/builds/117717662

CI loop fix needed?

**Comment by pascal-brand-st-dev** [comment]
You're right. Please ignore Travis failure. FYI, we will make an OP-TEE very soon, so I will check at your patch after the release is done - early next week.

**Comment by kuscsik** [comment]
Fixed the build issues. I could reproduce them with Clang based builds on Android 6.0.

**Comment by kuscsik** [comment]
I will push a new version. I just noticed that my tab/space indentation is wrong for this project.

File: libteec/src/teec_trace.c [hunk 1]
```diff
@@ -100,7 +100,7 @@ int _dprintf(const char *function, int flen, int line, int level,
 		func = function;
 #endif
 
-#ifdef ANDROID_BUILD
+#ifdef __ANDROID__
```
Comment (pascal-brand-st-dev) [inline libteec/src/teec_trace.c]: Can we remove the comment "TOD(emi)" below?

File: Android.mk [hunk 2]
```diff
@@ -9,11 +9,13 @@ LOCAL_PATH := $(call my-dir)
 include $(LOCAL_PATH)/config.mk
 include $(LOCAL_PATH)/flags.mk
 
+forced_optee_CFLAGS =  $(CFLAGS)  -Wno-missing-include-dirs
```
Comment (pascal-brand-st-dev) [inline Android.mk]: Can't we have `CFLAGS += -Wno-missing-include-dirs` instead?

File: Android.mk [hunk 3]
```diff
@@ -38,26 +38,4 @@ LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)/public
...
-
-LOCAL_C_INCLUDES := $(LOCAL_PATH)/public \
-    $(LOCAL_PATH)/libteec/include \
-    $(LOCAL_PATH)/tee-supplicant/src
-
-LOCAL_SHARED_LIBRARIES := libteec
-LOCAL_MODULE := tee-supplicant
-LOCAL_MODULE_TAGS := optional
-include $(BUILD_EXECUTABLE)
+include $(LOCAL_PATH)/tee-supplicant/tee_supplicant.mk
```
Comment (pascal-brand-st-dev) [inline Android.mk]: `Android.mk` or `tee_supplicant_android.mk` instead?

File: tee-supplicant/tee_supplicant.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                -DBINARY_PREFIX=\"TEES\" \
+                -DTEEC_LOAD_PATH=\"$(CFG_TEE_CLIENT_LOAD_PATH)\" \
+
+LOCAL_SRC_FILES += src/handle.c \
+                   src/tee_supp_fs.c \
+                   src/tee_supplicant.c \
+                   src/teec_ta_load.c \
+                   src/rpmb.c
+
+LOCAL_C_INCLUDES := $(LOCAL_PATH)/public \
```
Comment (pascal-brand-st-dev) [inline tee-supplicant/tee_supplicant.mk]: `$(LOCAL_PATH)/../public` instead?

File: tee-supplicant/src/tee_supplicant.c [hunk 5]
```diff
@@ -333,10 +333,10 @@ static int get_dev_fd(void)
 	return -1;
 }
 
-static void usage(void)
+static int usage(void)
 {
 	fprintf(stderr, "usage: tee-supplicant [<device-name>]");
-	exit(1);
+	return 1;
```
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: `return EXIT_FAILURE;` instead?

File: libteec/src/teec_trace.c (hunk 1 above)
Comment (kuscsik) [inline libteec/src/teec_trace.c]: Actually Android supports syscall() starting with Android 5.0. I suggest we remove this condition for Android builds.

File: Android.mk (hunk 2 above)
Comment (kuscsik) [inline Android.mk]: I don't think it is a good idea to carry around the CFLAGS variable in Android makefiles. It suggests that it is a global variable accepted by the Android builds system.
Having the forced_optee_CFLAGS variable defined makes the Android.mk more readable.

File: Android.mk (hunk 3 above)
Comment (kuscsik) [inline Android.mk]: Android.mk files are auto-included by the main android Makefile - which we don't want in this case. I will rename it to tee_supplicant_android.mk.

File: tee-supplicant/tee_supplicant_android.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+                -DBINARY_PREFIX=\"TEES\" \
+                -DTEEC_LOAD_PATH=\"$(CFG_TEE_CLIENT_LOAD_PATH)\" \
+
+LOCAL_SRC_FILES += src/handle.c \
+                   src/tee_supp_fs.c \
+                   src/tee_supplicant.c \
+                   src/teec_ta_load.c \
+                   src/rpmb.c
+
+LOCAL_C_INCLUDES := $(LOCAL_PATH)/public \
```
Comment (pascal-brand-st-dev) [inline tee-supplicant/tee_supplicant_android.mk]: `$(LOCAL_PATH)/../public` instead?

**Comment by pascal-brand-st-dev** [comment]
Apart from 1 comment about `../public`:
`Reviewed-by: Pascal Brand <pascal.brand@linaro.org>`

**Comment by kuscsik** [comment]
Fixed ../public comment from Pascal. Added Tested-by.

**Pull request merged**